atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/pool/CassandraService.java

```diff
@@ -84,8 +105,10 @@ public void close() {
             ImmutableRangeMap.Builder<LightweightOppToken, List<InetSocketAddress>> newTokenRing =
                     ImmutableRangeMap.builder();
 
-            // grab latest token ring view from a random node in the cluster
+            // grab latest token ring view from a random node in the cluster and update local hosts
             List<TokenRange> tokenRanges = getTokenRanges();
+            myLocation = new HostLocationSupplier(getSnitch(), myLocation).get();
```

Comment:
does this mean the default location changes every time you refresh token ranges? if so, then it's a bit strange to call it a default, it's more of an initial location?